tests/lib/nested.sh

```diff
@@ -212,9 +213,13 @@ refresh_to_new_core(){
             execute_remote "sudo snap refresh snapd --${NEW_CHANNEL}"
             execute_remote "snap info snapd" | grep -E "^tracking: +latest/${NEW_CHANNEL}"
         else
-            execute_remote "sudo snap refresh core --${NEW_CHANNEL}"
+            CHANGE_ID=$(execute_remote "sudo snap refresh core --${NEW_CHANNEL} --no-wait")
             wait_for_no_ssh
             wait_for_ssh
+            # wait for the refresh to be done before checking, if we check too 
+            # quickly then operations on the core snap like reverting, etc. may
+            # fail because it will have refresh-snap change in progress
+            execute_remote "snap watch $CHANGE_ID"
```

Comment:
I think `snap watch --last=refresh?` should work too, and we don't have to track CHANGE_ID anymore.. No strong opinion though.

Reply:
Yes, this should work.

Reply:
I'll leave it as is, I think being more explicit here is good